src/main/java/seedu/duke/operations/Storage.java

```diff
@@ -0,0 +1,143 @@
+package seedu.duke.operations;
+
+import seedu.duke.DukeException;
+import seedu.duke.task.DeadlineTask;
+import seedu.duke.task.Task;
+import seedu.duke.task.EventTask;
+import seedu.duke.task.TodoTask;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Storage provides a way for Duke to store and load TaskList in
+ * between sessions.
+ */
+public class Storage {
+    private final String filePath;
+
+    /**
+     * To initialize a Storage, a filepath to the data file is needed.
+     *
+     * @param filePath      Filepath to data file
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Returns the previous session's TaskList from data file indicated
+     * within Storage's filePath attribute.
+     *
+     * @return                  ArrayList of past saved tasks
+     * @throws DukeException    Thrown when Storage has issue loading data
+     */
+    public ArrayList<Task> load() throws DukeException {
+        File dataFile = new File(filePath);
+        if (!dataFile.exists()) {
+            setupDirectory();
+            return new ArrayList<>();
+        } else {
+            ArrayList<Task> loadedTasks = new ArrayList<>();
```

Comment:
Although type inference is valid it would be better to explicitly state types.
```suggestion
            return new ArrayList<Task>();
        } else {
            ArrayList<Task> loadedTasks = new ArrayList<Task>();
```